notebooks/contrib-dev/Fraction_of_Molecular_Collisions.ipynb

```diff
@@ -280,48 +266,47 @@
+        "## 2. Determining the Fraction of Molecules\n",
+        "Write a function that takes in an array of temperatures, a lower bound of integration (activation energy), and an upper bound of integration (infinity unless given) and returns the fraction of collisions that can overcome an activation energy.\n",
         "\n",
-        "Hint: Perform integration of $f(E,T)$ from 0 to 25 kcal?"
+        "*Hint: use integrate.quad. Look up documentation to see its output*"
       ]
     },
     {
       "cell_type": "code",
-      "execution_count": null,
+      "execution_count": 6,
       "metadata": {
         "id": "h6z_ScYVD1sj"
       },
       "outputs": [],
       "source": [
-        "# Regenerate an array of tempeartures for calculation based on the requirements\n",
+        "# Regenerate an array of temperatures for calculation based on the requirements\n",
         "\n",
-        "# Define a function for fraction of molecules that has sufficient energy to pass over an energy barrier calculation\n",
-        "def fraction(T):\n",
+        "# Define a function for fraction of molecules that have sufficient energy to pass over an energy barrier calculation\n",
+        "def fraction(T, low, upp):\n",
         "  \"\"\"Using integration to calculate the fraction of molecules from 0 to 25 kcal\n",
         "    Args:\n",
-        "        T: temperature (K)\n",
+        "        T: array of temperatures (K)\n",
+        "        low: lower bound of integration (minimum energy barrier)\n",
+        "        upp: upper bound of integration (infinity or a given value)\n",
         "        E: Energy (per mole)\n",
         "    Returns:\n",
-        "        f(E,T) : distribution of collision for energy per mole (mol/cal)\n",
+        "        f: f(E,T), distribution of collision for energy per mole (mol/cal)\n",
         "    \"\"\"\n",
         "\n",
-        "  # Store fraction of molecules that does not have sufficient energy to pass over an energy barrier of 25 kcal in a list\n",
-        "  I = []\n",
+        "  # Store fraction of molecules that have sufficient energy to pass over an energy barrier equal to low\n",
+        "  f = []\n",
         "\n",
-        "  # i in terms of the order of temperatures in T\n",
-        "  i = 0\n",
-        "   \n",
-        "  # Create a for loop for integration calculation at different temperatures\n",
         "  # Add your solution here\n",
         "  ### BEGIN SOLUTION\n",
+        "  # Create a for loop for integration calculation at different temperatures\n",
         "  for i in range(len(T)):\n",
-        "    result,error = integrate.quad(lambda E: 2*pi*(1/(pi*R*T[i]))**1.5*E**0.5*np.exp(-E/(R*T[i])),0,25)\n",
-        "    I.append(result)\n",
+        "    result,error = integrate.quad(lambda E: 2*pi*(1/(pi*R*T[i]))**1.5*E**0.5*np.exp(-E/(R*T[i])),low,upp)\n",
+        "    f.append(result)\n",
         "\n",
         "  ### END SOLUTION\n",
```

Comment:
Please check all "### END SOLUTION ###" are complete. This one is missing the trailing "###"